### PR TITLE
doc: enrich readme

### PR DESCRIPTION
**Pull request by gxthrj**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -17,18 +17,46 @@
...
-## what is apisix-control-plane?
-apisix-control-plane provide APISIX with a `yml` configuration capability,
-We can use `yml` to define the proxy behavior of APISIX
+## What is apisix-control-plane
 
-## Why do you want to do this?
-1. In order to facilitate the integration of k8s, use `yml` to define APISIX;
-2. For easier synchronization across clusters;
-3. Can be better adapted to multiple platforms (k8s, vm);
+apisix-control-plane is an implementation of providing a control plane for apache apisix. with a `yml` configuration capability, We can use `yml` to define the behavior of APISIX.
```
Comment (moonming) [inline README.md]: We not just want to support APISIX, and will support other DP

**Review by moonming — CHANGES_REQUESTED** [review]

File: README.md (hunk 1 above)
Comment (gxthrj) [reply README.md]: I wrote [here](https://github.com/apache/apisix-control-plane/pull/27/files#diff-04c6e90faac2675aa89e2176d2eec7d8R46) that can provide configuration capabilities for other gateways.
Let me add another explanation

File: README.md [hunk 2]
```diff
@@ -17,18 +17,48 @@
...
-## what is apisix-control-plane?
-apisix-control-plane provide APISIX with a `yml` configuration capability,
-We can use `yml` to define the proxy behavior of APISIX
+## What is apisix-control-plane
 
-## Why do you want to do this?
-1. In order to facilitate the integration of k8s, use `yml` to define APISIX;
-2. For easier synchronization across clusters;
-3. Can be better adapted to multiple platforms (k8s, vm);
+apisix-control-plane is an implementation of providing a control plane for DPs (Data Plane) like apache apisix or other gateways. with a `yml` configuration capability, We can use `yml` to define the behavior of any DP.
```
Comment (membphis) [inline README.md]: Is `yaml` easier to understand?

File: README.md [hunk 3]
```diff
@@ -17,18 +17,48 @@
...
+
+## Why do we need to implement a control-plane
+
+for now, we have a default implementation for apache APISIX.
```
Comment (membphis) [inline README.md]: Capitalize the first letter

File: README.md [hunk 4]
```diff
@@ -17,18 +17,48 @@
...
+
+As we knows, apache APISIX is a high-performance gateway. When using APISIX we need to inform apisix of some proxy rules, and distribute these rules to apisix modules. We can call it contol-plane, similar to pilot in istio. Of course, the distribution configuration is only the most basic function of the control plane.
```
Comment (membphis) [inline README.md]: `apache` -> `Apache`

File: README.md [hunk 5]
```diff
@@ -17,18 +17,48 @@
...
+
+We know that apisix already has admin api, so why do we need to implement a control-plane?
```
Comment (membphis) [inline README.md]: `admin api` -> `Admin API`, is it better?

File: README.md [hunk 6]
```diff
@@ -17,18 +17,48 @@
...
+
+In fact, leaving aside the admin api, what we need is a description method that can completely define the rules while maintaining the legibility of the admin api. Therefore, we have [discussion here](https://github.com/apache/apisix-control-plane/blob/master/doc/yaml_struct.md), and we have implemented a version of the basic functions according to this data structure , I hope everyone puts forward their views. We can also submit a PR to modify this [document](doc/yaml_struct.md).
+
+Secondly, apache apisix admin api uses id associations to strongly bind objects to express the relationship between objects. For example, the route object uses service_id and upstream_id to bind the relationship with service and upstream respectively.
+
+This kind of representation, like the well-known relational database, is easier to understand and reduces redundant configurations. But it also brings some other problems. For example, rely on verification. When we want to bind an `upstreamId` to a route, we must first create the upstream, and then bind the generated `upstreamId` to the route. This process corresponds to two admin APIs, and there is no transaction management.
```
Comment (membphis) [inline README.md]: `upstreamId` -> `upstream_id`, keep the same as APISIX implementation.

File: README.md [hunk 7]
```diff
@@ -17,18 +17,48 @@
...
+
+We hope that the emergence of control-plane can solve the above two problems.
```
Comment (membphis) [inline README.md:42]: > We hope that the emergence of control-plane can solve the above two problems.

Here is another style, I think it is easier for reading.

```
We hope that the emergence of control-plane can solve the above two problems.

1. ****
2. ****
```

File: README.md [hunk 8]
```diff
@@ -17,18 +17,48 @@
...
+For example, based on gitops configuration version management, it is also more convenient to use yaml to define apisix in k8s.
+
+Realize the prototype of cross-cluster synchronization configuration through the control panel. In the future, we can also enrich the functions of the control plane and simplify management under the cluster mode of multi-platform hybrid deployment (k8s/vm). We can also shield specific implementations of APISIX and provide configuration functions for more types of gateways.
+
+## Feature support
+
+1. Support the declarative definition of yaml.
+2. Use the memory database to synchronize the gateway status.
+3. Diff capabilities based on memory objects.
+4. sync / update the apisix configuration.
```
Comment (membphis) [inline README.md]: Capitalize the first letter

File: README.md [hunk 9]
```diff
@@ -17,18 +17,50 @@
...
+
+## Feature support
+
+1. Support the declarative definition of yaml.
+2. Use the memory database to synchronize the gateway status.
+3. Diff capabilities based on memory objects.
+4. Sync / update the apisix configuration.
+5. Support incremental synchronization.
+6. Support transaction processing.
+
```
Comment (moonming) [inline README.md:60]: What's the plan for other DPs?

File: README.md [hunk 10]
```diff
@@ -21,25 +21,27 @@
...
+apisix-control-plane is an implementation of providing a control plane for DPs (Data Plane) like Apache APISIX or other gateways. with a `yaml` configuration capability, We can use `yaml` to define the behavior of any DP.
 
 ## Why do we need to implement a control-plane
 
-for now, we have a default implementation for apache APISIX.
+For now, we have a default implementation for Apache APISIX.
 
-As we knows, apache APISIX is a high-performance gateway. When using APISIX we need to inform apisix of some proxy rules, and distribute these rules to apisix modules. We can call it contol-plane, similar to pilot in istio. Of course, the distribution configuration is only the most basic function of the control plane.
+As we knows, Apache APISIX is a high-performance gateway. When using APISIX we need to inform apisix of some proxy rules, and distribute these rules to apisix modules. We can call it contol-plane, similar to pilot in istio. Of course, the distribution configuration is only the most basic function of the control plane.
```
Comment (membphis) [inline README.md]: `When using APISIX we need to inform apisix`

Please pay attention to "apisix", they are wrong styles.

**Review by membphis — APPROVED** [review]
LGTM, except for some minor doc style.

File: README.md (hunk 9 above)
Comment (gxthrj) [reply README.md:60]: Added some DP plans below

File: README.md (hunk 10 above)
Comment (gxthrj) [reply README.md]: done

**Review by moonming — APPROVED** [review]

**Pull request merged**